llvm/test/CodeGen/RISCV/cheri/rematerialize.mir

```diff
@@ -0,0 +1,113 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
+# Check that we can rematerialize constant CIncOffset/CMove instead of spilling to the stack
+# RUN: %riscv64_cheri_llc -verify-machineinstrs %s -o - -run-pass=simple-register-coalescing | FileCheck %s
+--- |
+  target datalayout = "e-m:e-pf200:128:128:128:64-p:64:64-i64:64-i128:128-n64-S128-A200-P200-G200"
+  target triple = "riscv64-unknown-freebsd"
+
+  declare void @external_fn(i8 addrspace(200)*) addrspace(200) #0
+
+  ; Function Attrs: nounwind
+  define void @test_remat() addrspace(200) #2 {
+    call void @external_fn(i8 addrspace(200)* getelementptr (i8, i8 addrspace(200)* null, i64 123)) #3
+    ret void
+  }
+
+  attributes #0 = { "target-features"="+cap-mode,+xcheri" }
+  attributes #1 = { nounwind readnone willreturn "target-features"="+cap-mode,+xcheri" }
+  attributes #2 = { nounwind "target-features"="+cap-mode,+xcheri" }
+  attributes #3 = { nounwind }
```

Comment:
```suggestion
  declare void @external_fn(i8 addrspace(200)*) addrspace(200)

  define void @test_remat() addrspace(200) no unwind {
    call void @external_fn(i8 addrspace(200)* getelementptr (i8, i8 addrspace(200)* null, i64 123))
    ret void
  }

```
is hopefully all that's necessary (maybe also needs a `nounwind` on the `declare`, but shouldn't need anything else).

Reply:
Yeah I don't think it looks at it at all.

Reply:
Even for regenerating the MIR it should be sufficient.